File: android/app1mvvm/app/src/main/java/edu/curso/app1_mvvm/MainActivity.java
package edu.curso.app1_mvvm;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.BindingAdapter;
import androidx.databinding.DataBindingUtil;
import edu.curso.app1_mvvm.databinding.ContatoActivityBinding;
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ContatoActivityBinding activityMainBinding =
                DataBindingUtil.setContentView(this, R.layout.contato_activity);
        activityMainBinding.setViewModel( new ContatoViewModel() );
        activityMainBinding.executePendingBindings();
    }

    @BindingAdapter({"toastMessage"})
    public static void runMe(View view, String message) {
        if (message != null) {
            Toast.makeText(view.getContext(), message, Toast.LENGTH_LONG).show();
        }
    }
}

File: introduction-android-mobile-dev/VideoPlayer/app/src/main/java/edu/curso/videoplayer/MainActivity.kt
package edu.curso.videoplayer

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.net.Uri;
import android.net.Uri.*;
import android.widget.MediaController;
import android.widget.VideoView;
import edu.curso.videoplayer.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        resources.openRawResource(R.raw.johann_pachelbe_canon)

        val mediaController = MediaController(this)
        binding.apply {
            mediaController.setAnchorView(videoView)
            val uri : Uri = parse ("android.resource://" + packageName +
                    "/raw/johann_pachelbe_canon")
            videoView.setMediaController(mediaController)
            videoView.setVideoURI(uri)
            videoView.requestFocus()
            videoView.start()
        }

    }
}